contracts/finance/VestingWalletConfidential.sol

```diff
@@ -53,7 +53,7 @@ abstract contract VestingWalletConfidential is OwnableUpgradeable, ReentrancyGua
     }
 
     /// @dev Amount of token already released
-    function released(address token) public view virtual returns (euint64) {
+    function released(address token) public view virtual returns (euint128) {
         return _getVestingWalletStorage()._tokenReleased[token];
     }
 
```

Comment:
And how `FHE.asEuint64(..)` is safe here?
https://github.com/OpenZeppelin/openzeppelin-confidential-contracts/blob/498c67cea051562c2d5a01faa3dec4eec9309fb8/contracts/finance/VestingWalletConfidential.sol#L60-L66

Reply:
total supply is `euint64`, so if the releasable amount is over that, something is very wrong. The result of the subtraction is always under `euint64`.max